src/renderer.c

```diff
@@ -372,30 +499,48 @@ int ren_font_group_get_height(RenFont **fonts) {
 double ren_font_group_get_width(RenFont **fonts, const char *text, size_t len, int *x_offset) {
   double width = 0;
   const char* end = text + len;
-  GlyphMetric* metric = NULL; GlyphSet* set = NULL;
+
   bool set_x_offset = x_offset == NULL;
   while (text < end) {
     unsigned int codepoint;
     text = utf8_to_codepoint(text, &codepoint);
-    RenFont* font = font_group_get_glyph(&set, &metric, fonts, codepoint, 0);
-    if (!metric)
-      break;
-    width += (!font || metric->xadvance) ? metric->xadvance : fonts[0]->space_advance;
-    if (!set_x_offset) {
+    GlyphMetric *metric = NULL;
+    font_group_get_glyph(fonts, codepoint, 0, NULL, &metric);
+    width += is_whitespace(codepoint) || !metric || !metric->xadvance // important: fonts can provide xadvances for whitespaces
+      ? fonts[0]->space_advance * (float) (codepoint == '\t' ? fonts[0]->tab_size : 1) // tabs are multiplied by tab size
+      : metric->xadvance;
```

Comment:
Make this into a function/macro, as it's also used in
https://github.com/lite-xl/lite-xl/blob/1f1c33947fcdb0c6543f73d9aa7a2549ab7d2264/src/renderer.c#L617-L619